doc/nrf/libraries/others/audio_module.rst

```diff
@@ -0,0 +1,120 @@
+.. _lib_audio_module:
+
+Audio module
+############
+
+.. contents::
+   :local:
+   :depth: 2
+
+The audio module library is an interface to audio processing functions that coordinate audio states and the exchange of audio-related data of an LE Audio application, such as :ref:`nrf5340_audio`.
+
+Overview
+********
+
+The audio module is an interface for constructing custom audio processing modules, such as decoder, encoder, and I2S output.
+It provides a common interface to audio processing algorithms.
+
+Using this interface, you can open and configure the custom modules, connect to them, and start and stop them.
+You can also send audio data to and from the application.
+
+The operation of the module is determined by a set of functions that perform the processing.
+Connecting these modules lets you create the following audio system or stream:
+
+.. figure:: images/audio_module_stream.svg
+   :alt: Audio stream example
+
+Implementation
+==============
+
+The audio module is implemented as a set of functions, listed in the following figure:
+
+.. figure:: images/audio_module_functions.svg
+   :alt: Audio module functions
+
+These functions call out to the user's implementation of these functions written to a predetermined API for the desired audio algorithm.
+
+The following table outlines the available functions that are defined in :c:struct:`struct audio_module_functions` and whether they are mandatory or not:
+
+.. list-table::
+    :header-rows: 1
+
+    * - Function
+      - Mandatory/Optional
+      - Comment
+    * - :c:member:`audio_module_functions.*open`
+      - Optional
+      - Perform any algorithm operations to open.
+    * - :c:member:`audio_module_functions.*close`
+      - Optional
+      - Tidy up the algorithm on a close.
+    * - :c:member:`audio_module_functions.*configure_set`
+      - Mandatory
+      - Configure the algorithm to perform a particular operation.
+    * - :c:member:`audio_module_functions.*configuration_get`
+      - Mandatory
+      - Return the internal configuration of the algorithm.
+    * - :c:member:`audio_module_functions.*start`
+      - Optional
+      - Set the algorithm running.
+    * - :c:member:`audio_module_functions.*stop`
+      - Optional
+      - Stop the algorithms.
+    * - :c:member:`audio_module_functions.*data_process`
+      - Mandatory
+      - Process the data for the algorithm.
```

Comment:
```suggestion
    * - ``audio_module_functions.*open``
      - Optional
      - Perform any algorithm operations to open.
    * - ``audio_module_functions.*close``
      - Optional
      - Tidy up the algorithm on a close.
    * - ``audio_module_functions.*configure_set``
      - Mandatory
      - Configure the algorithm to perform a particular operation.
    * -``audio_module_functions.*configuration_get``
      - Mandatory
      - Return the internal configuration of the algorithm.
    * - ``audio_module_functions.*start``
      - Optional
      - Set the algorithm running.
    * - ``audio_module_functions.*stop``
      - Optional
      - Stop the algorithms.
    * - ``audio_module_functions.*data_process``
      - Mandatory
      - Process the data for the algorithm.
```
Let's keep it simple to make the doc build green. If I find a better solution, I'll open a separate PR or add a new comment.